### PR TITLE
Feat/calico-3.23.2

### PR DESCRIPTION
**Pull request by ralgozino**
update calico to 3.23.2: fixes stale tokens issue with Kubernetes >= v1.21
update docs to prepare for new release

### CODE REVIEW

File: katalog/calico/README.md [hunk 1]
```diff
@@ -7,43 +7,42 @@ Calico supports a broad range of platforms including Kubernetes, OpenShift, Dock
...
   - [https://github.com/projectcalico/cni-plugin](https://github.com/projectcalico/cni-plugin).
-  - [https://github.com/projectcalico/pod2daemon](https://github.com/projectcalico/pod2daemon).
   - [https://github.com/projectcalico/node](https://github.com/projectcalico/node).
 
 ## Requirements
 
-- Tested with Kubernetes >= `1.18.X`.
-- Tested with Kustomize = `v3.3.X`.
-- Prometheus Operator.
+- Tested with Kubernetes >= `1.20.X`.
```
Comment (nutellinoit) [inline katalog/calico/README.md:33]:
```suggestion
- Tested with Kubernetes >= `1.21.X`.
```

File: katalog/calico/deploy.yml [hunk 2]
```diff
@@ -25,31 +28,34 @@ spec:
...
-          initialDelaySeconds: 10
-          failureThreshold: 6
-        readinessProbe:
-          exec:
-            command:
-            - /usr/bin/check-status
-            - -r
-          periodSeconds: 10
+        - name: calico-kube-controllers
+          image: docker.io/calico/kube-controllers:v3.23.2
```
Comment (nutellinoit) [inline katalog/calico/deploy.yml:40]: Maybe we can drop the tag here

File: katalog/calico/ds.yml [hunk 3]
```diff
@@ -26,147 +30,167 @@ spec:
...
-        volumeMounts:
-        - name: flexvol-driver-host
-          mountPath: /host/driver
-        securityContext:
-          privileged: true
+        # This container performs upgrade from host-local IPAM to calico-ipam.
+        # It can be deleted if this is a fresh installation, or if you have already
+        # upgraded to use calico-ipam.
+        - name: upgrade-ipam
+          image: docker.io/calico/cni:v3.23.2
```
Comment (nutellinoit) [inline katalog/calico/ds.yml:51]: Maybe we can drop the tag here

File: katalog/calico/ds.yml [hunk 4]
```diff
@@ -26,147 +30,167 @@ spec:
...
+            - mountPath: /var/lib/cni/networks
+              name: host-local-net-dir
+            - mountPath: /host/opt/cni/bin
+              name: cni-bin-dir
+          securityContext:
+            privileged: true
+        # This container installs the CNI binaries
+        # and CNI network config file on each node.
+        - name: install-cni
+          image: docker.io/calico/cni:v3.23.2
```
Comment (nutellinoit) [inline katalog/calico/ds.yml:78]: Maybe we can drop the tag here

**Review by nutellinoit — CHANGES_REQUESTED** [review]
I left some comments. The overall PR is 👍

File: README.md [hunk 5]
```diff
@@ -36,32 +38,34 @@ Click on each package to see its full documentation.
 
 ## Compatibility
 
-| Kubernetes Version |   Compatibility    |                        Notes                        |
+| Kubernetes Version |   Compatibility    | Notes                                               |
 | ------------------ | :----------------: | --------------------------------------------------- |
-| `1.20.x`           | :white_check_mark: | No known issues                                     |
+| `1.20.x`           |        :x:         | Incompatible                                        |
```
[the remark this answers is not in the document]
Comment (ralgozino) [reply README.md:43]: Explicit is better than implicit?  I'm fine either way, they info is in the comp matrix also

File: katalog/calico/README.md (hunk 1 above)
Comment (ralgozino) [reply katalog/calico/README.md:33]: nice catch, thanks!

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (nutellinoit) [reply README.md:43]: i think this little table must show only "what is working"

File: katalog/calico/deploy.yml (hunk 2 above)
Comment (ralgozino) [reply katalog/calico/deploy.yml:40]: I prefer to keep the manifests aligned with upstream (i.e. with the tags), it makes the maintenance easier and lets you spot changes in upstream.

Are there any downsides that I am missing?

(commenting just here for all the other tags, will update all of them with the result of this discussion)

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (ralgozino) [reply README.md:43]: after getting some sleep I remembered that, when we designed the new readme, the idea behind this section was to remove the whole compatibility matrix from here (that was too much) and show only the part that is currently under support (current K8s - 3), being that 1.23 is tech preview, following that logic the table should include 1.20. I actually think is the right approach.

[the remark this answers is not in the document]
Comment (nutellinoit) [reply README.md:43]: ok 👍

File: katalog/calico/deploy.yml (hunk 2 above)
Comment (ralgozino) [reply katalog/calico/deploy.yml:40]: gentle ping :-)

**Review by nutellinoit — APPROVED** [review]
LGTM 🎉

File: katalog/calico/deploy.yml (hunk 2 above)
Comment (ralgozino) [reply katalog/calico/deploy.yml:40]: deleted the duplicated tag from the `kustomization.yaml` file.

**Pull request merged**